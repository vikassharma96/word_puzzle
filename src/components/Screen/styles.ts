import {StyleSheet} from 'react-native';
import colors from '../../styles/colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.default.white,
  },
});

export default styles;
