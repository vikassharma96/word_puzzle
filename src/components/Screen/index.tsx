import React, {FC, ReactNode} from 'react';
import {SafeAreaProvider, SafeAreaView} from 'react-native-safe-area-context';
import styles from './styles';

type IProps = {children?: ReactNode};

const Screen: FC<IProps> = props => {
  return (
    <SafeAreaProvider>
      <SafeAreaView style={styles.container} edges={['top', 'left', 'right']}>
        {props.children}
      </SafeAreaView>
    </SafeAreaProvider>
  );
};

export default Screen;
