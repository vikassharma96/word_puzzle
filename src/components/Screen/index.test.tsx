import React from 'react';
import renderer from 'react-test-renderer';
import Screen from '.';

describe('@components/Screen', () => {
  const component = <Screen>{<></>}</Screen>;

  it('should match snapshots', () => {
    const componentTree = renderer.create(component).toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
