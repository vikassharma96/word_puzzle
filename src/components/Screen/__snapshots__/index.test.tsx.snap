// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Screen should match snapshots 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
